src/game_api/entities_items.hpp

```diff
@@ -91,12 +99,22 @@ class VladsCape : public Cape
 
 class Purchasable : public Movable
 {
-    virtual void buy(Entity* who) = 0;
+  public:
+    /// Be aware that for pickable items like weapons, the item will be automatically picked up by the player (unless he already holds something), the ownership transfer is a little bit slow, so it might trigger the shop keeper if you're not inside the shop.
+    /// does not consume money
+    virtual void acquire(Entity* who) = 0;
```

Comment:
Btw. no objections to this name? I didn't like the "buy" name since it does not deal with the money and the ownership is janky as for the comment

Reply:
ah, I see, seems to be the function that gets called after the item is bought, to make the player to grab the item if possible (grab item, equip backitem), the comment for me made it seem like its a side effect of the function, rather than it's main thing.
And for the name I'm not sure, I guess it's fine as long as the doc comments is good enough? I think it's a little ambiguous but I'm not gonna be perfectionist, it could be fine as that if you also think so, and we don't come up with any better name

Reply:
Testing a bit more, for powerup pickupables, calling it directly seems to just grant the powerup to the entity, just like being bought for free, same with backitems. But for items like guns, it just makes the player to grab the entity without doing anything else

Reply:
If i remember from testing, it removed the gun from the shop items.

Reply:
ok, so ... I think it's still valid function to detect purchase. Thou needs better comment and maybe better name?

Reply:
if you don't wanna include the purchase action in it: `equip`, `try_equip`, `try_auto_equip`

Reply:
the `on_` are usually problematic, as the hooks for me look stupid as `set_pre_on_something`